packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx

```diff
@@ -693,25 +700,40 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
     const { xBarScale, yBarScale } = this._getScales(containerHeight, containerWidth);
     const colorScale = this._createColors();
 
+    const baselinePoint = this._yMax < 0 ? this._yMax : 0;
     const bars = this._points.map((point: IVerticalBarChartDataPoint, index: number) => {
       const shouldHighlight = this._legendHighlighted(point.legend!) || this._noLegendHighlighted() ? true : false;
       this._classNames = getClassNames(this.props.styles!, {
         theme: this.props.theme!,
         legendColor: this.state.color,
         shouldHighlight,
       });
-      const barHeight: number = Math.max(yBarScale(point.y), 0);
-      let adjustedBarHeight = 0;
 
-      if (barHeight <= 0) {
+      let barHeight: number = this.props.supportNegativeYValues
+        ? yBarScale(point.y) - yBarScale(baselinePoint)
+        : yBarScale(point.y);
+      const isHeightNegative = barHeight < 0;
+      barHeight = Math.abs(barHeight);
+      // Calculate threshold for minimum visible bar height
+      const maxHeightFromBaseline =
+        this._yMax < 0 ? Math.abs(this._yMin - baselinePoint) : Math.abs(this._yMax - baselinePoint);
```

Comment:
>this._yMin - baselinePoint) [](http://example.com/codeflow?start=34&length=27)

this should be max of (abs(ymin - baseline), abs(ymax - baseline))

Reply:
for the case this._ymax<0 , abs(ymin) will always be greater than abs(ymax). Hence we don't need max for this.
But yes, for other case, we must take that.